app/dashboard/src/appUtils.tsx

```diff
@@ -1,53 +1,64 @@
 /** @file Constants related to the application root component. */
+import type { DriveCategory } from '#/layouts/CategorySwitcher/Category'
+import type { SettingsTabType } from '#/pages/dashboard/Settings/settingsData'
 
 // =================
 // === Constants ===
 // =================
 
-// === Paths ===
-
-/** Path to the root of the app (i.e., the Cloud dashboard). */
-export const DASHBOARD_PATH = '/'
-/** Path to the login page. */
-export const LOGIN_PATH = '/login'
-/** Path to the registration page. */
-export const REGISTRATION_PATH = '/registration'
-/** Path to the confirm registration page. */
-export const CONFIRM_REGISTRATION_PATH = '/confirmation'
-/** Path to the page in which a user can restore their account after it has been
- * marked for deletion. */
-export const RESTORE_USER_PATH = '/restore-user'
-/** Path to the forgot password page. */
-export const FORGOT_PASSWORD_PATH = '/forgot-password'
-/** Path to the reset password page. */
-export const RESET_PASSWORD_PATH = '/password-reset'
-/** Path to the set username page. */
-export const SET_USERNAME_PATH = '/set-username'
-/** Path to the offline mode entrypoint. */
-/** Path to page in which the currently active payment plan can be managed. */
-export const SUBSCRIBE_PATH = '/subscribe'
-export const SUBSCRIBE_SUCCESS_PATH = '/subscribe/success'
+export const SEARCH_PARAMS_PREFIX = 'cloud-ide_'
+
+const APP_PATHS = [
+  'drive',
+  'editor',
+  'settings',
+  'login',
+  'registration',
+  'confirmation',
+  'restore-user',
+  'forgot-password',
+  'password-reset',
+  'set-username',
+  'subscribe',
+  'subscribe/success',
+] as const
+
+/** Valid paths for a page in the app. */
+export type AppPath =
+  | Exclude<(typeof APP_PATHS)[number], 'editor' | 'settings'>
+  | `drive/${DriveCategory}`
+  | `editor/${string}`
+  | `settings/${SettingsTabType}`
+
+/** Valid paths to a page in the app, including the leading slash and the search query
+ * (if any). */
+export type AppFullPath = `/${AppPath}${'' | `?${string}`}`
+
+/** A type-safe function to create a path. */
+export function definePath(path: AppPath) {
+  return path
+}
+
 /** A {@link RegExp} matching all paths. */
 export const ALL_PATHS_REGEX = new RegExp(
-  `(?:${DASHBOARD_PATH}|${LOGIN_PATH}|${REGISTRATION_PATH}|${CONFIRM_REGISTRATION_PATH}|` +
-    `${FORGOT_PASSWORD_PATH}|${RESET_PASSWORD_PATH}|${SET_USERNAME_PATH}|${RESTORE_USER_PATH}|` +
-    `${SUBSCRIBE_PATH}|${SUBSCRIBE_SUCCESS_PATH})$`,
+  `/$|/(?:${APP_PATHS.map((path) => (path === 'editor' ? `${path}/` : path)).join('|')})\\b.*?$`,
```

Comment:
Isn't this more permissive since it matches things like `/registrationblah`, which wouldn't be accepted previously? Is this intentional?

Reply:
shouldn't be because of the trailing `\\b` - however it would technically accept trailing segments which i *think* is fine? its main function is as a heuristic to guess what path is the root path of the app

Reply:
<img width="489" alt="image" src="https://github.com/user-attachments/assets/86aada84-8958-4f0b-a7f2-5689a1b21607">

Ah yeah I was testing with a `/` in the path.